bindings-go/apis/v2/componentdescriptor.go

```diff
@@ -427,3 +437,35 @@ func (o *ComponentReference) GetIdentity() Identity {
 func (o *ComponentReference) GetIdentityDigest() []byte {
 	return o.GetIdentity().Digest()
 }
+
+// DigestSpec defines the digest and algorithm.
+// +k8s:deepcopy-gen=true
+// +k8s:openapi-gen=true
+type DigestSpec struct {
+	Algorithm string `json:"algorithm"`
+	Value     string `json:"value"`
+}
+
+// SignatureSpec defines the signature and algorithm.
+// +k8s:deepcopy-gen=true
+// +k8s:openapi-gen=true
+type SignatureSpec struct {
+	Algorithm string `json:"algorithm"`
+	Data      string `json:"data"`
+}
+
+type NormalisationVersion string
+
+const (
+	NormalisationVersionV1 NormalisationVersion = "v1"
+)
+
+// Signature defines a digest and corresponding signature, identifyable by name.
+// +k8s:deepcopy-gen=true
+// +k8s:openapi-gen=true
+type Signature struct {
+	Name                 string               `json:"name"`
+	NormalisationVersion NormalisationVersion `json:"normalisationVersion"`
```

Comment:
can you add an explenation where this version is used or when we should increase this version.

E.g. if its just the version of the json format then we should also name the signtaure accordingly like `SingatureV1`

Reply:
NormalisationVersion refers to the json format we use. So to validate a signature, we need to know the name, how the normalised descriptor was created, the digest and signature algorithm.

You suggest adding a version field to use one name with multiple versions? Or do you suggest an implicit namingSchema mySignatureV1?